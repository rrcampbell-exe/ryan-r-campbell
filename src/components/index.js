import BlogTags from './Blog/BlogTags'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Contact from './Contact/Contact'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'
import Loading from './Loading/Loading'
import Quote from './Quote/Quote'
import TextAndImage from './TextAndImage/TextAndImage'
import HamburgerMenu from './HamburgerMenu/HamburgerMenu'
import PopText from './PopText/PopText'
import PodcastTeaser from './Podcast/PodcastTeaser'
import Boasts from './Tech/Boasts'
import SkillsCloud from './Tech/SkillsCloud'
import BlogFooter from './Blog/BlogFooter'
import SubstackRedirect from './SubstackRedirect/SubstackRedirect'
import { CodeSpan, CodeBlock } from './CodeBlock/CodeBlock'

export { BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner, Loading, Quote, TextAndImage, BlogTags, HamburgerMenu, PopText, PodcastTeaser, Boasts, SkillsCloud, BlogFooter, CodeSpan, CodeBlock, SubstackRedirect }
