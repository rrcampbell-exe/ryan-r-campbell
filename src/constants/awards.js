const awards = [
  { 
    name: '2022 International Book Awards Finalist - Humor and Satire',
    alt: "A wreath containing the words International Book Awards Finalist - 2022 | Humor & Satire",
    link: 'https://www.internationalbookawards.com/2022pressrelease.html',
    src: './awards/2022-international-book-awards-finalist-humor-and-satire.png',
    className: 'light award',
  },
  {
    name: "Readers' Favorite Five Stars",
    alt: "The logo for the Readers' Favorite Five Stars award, which is a silver medallion that features five stars",
    link: 'https://readersfavorite.com/book-review/scambait',
    src: './awards/readers-favorite-five-stars.png',
    className: 'award',
  },
  {
    name: '2020 International Book Awards Finalist - Science Fiction',
    alt: "A wreath containing the words International Book Awards Finalist - 2020 | Science Fiction",
    link: 'https://ryanrcampbell.com/2020/07/23/imminent-dawn-named-as-finalist-in-2020-international-book-awards/',
    src: './awards/2020-international-book-awards-finalist-science-fiction.png',
    className: 'light award',
  }
]

export default awards
