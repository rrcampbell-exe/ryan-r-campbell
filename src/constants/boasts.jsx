import { BrowserSVG, LaptopSVG, ManagementSVG } from '../assets/svg'

const boasts = [
  {
    category: 'Hackathons',
    description: [
      'Finalist in 2022 Hack for Good with augmented reality project to ease treatment for pediatric cancer patients', 
      'Awarded "Most Innovative" in 2024 Developer Experience Hackathon for AI-powered sentiment analysis application'
    ],
    image: <LaptopSVG />
  },
  {
    category: 'Leadership & Culture',
    description: [
      'Founded company-internal podcast to better connect teams and discuss pressing technical topics',
      'Presented on organizational impact of well-crafted mission statements', 
      'Frequent contributor to company blog on topics related to sharpening technical and non-technical skills', 
      'Developed documentation for and trained others on Kubernetes namespace migration'
    ],
    image: <ManagementSVG />
  },
  {
    category: 'UI/UX', 
    description: ['Created React library of reusable components to streamline implementation of common design patterns', 
      'Led team initiative to migrate to modern enterprise design system', 
      'Contributed directly to the codebases of bandwidth-strained teams to squash bugs and improve client experience',
      'Exceeded organizational goal of 60% unit testing coverage by achieving 93.9% and 82.7% coverage for new applications' 
    ],
    image: <BrowserSVG />
  }
]

export default boasts
