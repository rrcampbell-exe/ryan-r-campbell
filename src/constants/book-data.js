const books = [
  {
    title: 'Scambait',
    genre: 'Contemporary Fiction | Humor',
    cover: './books/scambait/scambait-flat-cover-international-book-awards-readers-favorite-sticker.png',
    alt: 'Orange cover of the book Scambait by Ryan R. Campbell, which displays the book title and shows the main character sitting at a desk with a computer and his Saint Bernard at his side',
    summary: 'The International Book Awards finalist in Humor and Satire that asks, "Who scams the scammers?"',
    fullDescription: {
      paragraphOne: 'Ever since his grandma squandered the family fortune to two-bit hucksters from “Microsoft Tech Support,” corporate do-little Eric Amundsen has spent his workdays wasting the time of Nigerian princes and car warranty salesmen everywhere.',
      paragraphTwo: `But when Eric's supposedly long-deceased father contacts him through his spam folder, Eric—whose workday mantra isn't exactly “solutions-oriented”—has serious problems to solve.`
    },
    readMoreLink: './books/scambait/scambait-teaser-chapters.pdf',
    buyLink: 'https://www.amazon.com/dp/B09SY4P6FR',
    shortPromoDescription: 'The 2022 International Book Awards Finalist in Humor and Satire.',
    id: 'scambait'
  },
  {
    title: 'And Ampersand',
    genre: 'Works of Short Fiction',
    cover: './books/and-ampersand/and-ampersand-flat-cover.png',
    alt: 'The black cover of the book And Ampersand by Ryan R. Campbell, which features two large ampersands in gold and white, respectively',
    summary: 'A collection of short fiction that begins with endings, ends with beginnings, and elevates feel over form.',
    fullDescription: {
      paragraphOne: 'Your death comes on slowly, while elsewhere, a nameless man fakes his own. Encased in glass, a young man navigates a world meant for those beyond. A mudslide washes away a once romantic weekend, and a childhood swim drowns all illusion of innocence.',
      paragraphTwo: `These are but some of the tales in this collection of flash fiction, short stories, and freeform prose.`
    },
    readMoreLink: './books/and-ampersand/glassed-excerpt-from-and-ampersand.pdf',
    buyLink: 'https://www.amazon.com/dp/1736387111',
    shortPromoDescription: "The perfect pocket companion for any reader who's tussled with the tie between beginning and end.",
    id: 'and-ampersand'
  },
  {
    title: 'Imminent Dawn',
    genre: 'Science Fiction',
    cover: './books/imminent-dawn/imminent-dawn-flat-cover-international-book-awards-sticker.png',
    alt: 'The blue cover of the book Imminent Dawn by Ryan R. Campbell, which shows the silhouette of a woman against a backdrop of stars and circuitry',
    summary: 'The International Book Awards finalist and first installment in the EMPATHY sci-fi saga.',
    fullDescription: {
      paragraphOne: 'When an art-school dropout enrolls in the first round of human trials for an internet-access brain implant, she never expects to find herself at the center of conspiracies stretching from the highest levels of the North American Union government to the lowest dredges of its shadowy hacking collectives.',
      paragraphTwo: ''
    },
    readMoreLink: './books/imminent-dawn/imminent-dawn-sample-chapter.pdf',
    buyLink: 'https://www.amazon.com/dp/B07N1Q8C6G',
    shortPromoDescription: 'The International Book Awards finalist and first installment in the EMPATHY sci-fi saga.',
    id: 'imminent-dawn'
  },
  {
    title: 'Mourning Dove',
    genre: 'Science Fiction',
    cover: './books/mourning-dove/mourning-dove-flat-cover.png',
    alt: 'The golden cover of the book Mourning Dove by Ryan R. Campbell, which features the silhouettes of a woman with a backpack and a cat set against a technothriller-like web of connected dots',
    summary: 'An evocative, sweeping symphony of love, revenge, and desperation in cacophonous times.',
    fullDescription: {
      paragraphOne: 'In the aftermath of the Human/Etech research study, spouse is further pit against spouse, brother against sister, and governments against their people. In the end, all must choose between reclaiming the past or surrendering to an intractable world of their own creation.',
      paragraphTwo: ''
    },
    readMoreLink: './books/mourning-dove/mourning-dove-sample-chapter.pdf',
    buyLink: 'https://www.amazon.com/dp/B07QZGSDX6',
    shortPromoDescription: 'The sequel to International Book Awards finalist, Imminent Dawn.',
    id: 'mourning-dove'
  },
]

export default books
