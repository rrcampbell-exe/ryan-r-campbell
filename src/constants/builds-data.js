const builds = [
  {
    logo: './builds/real-fake-birds-app-logo.png',
    alt: 'The cartoon outline of a bird against a forest green background',
    title: 'Real Fake Birds',
    teaser: 'Some are real. Some are fake. Tell them apart.',
    description: 'Earn badges and extend your streak in this real-or-fake guessing game that puts even the most ardent birders to the test.',
    technologies: 'React, Node, axios, styled-components, react-router, REST APIs',
    link: 'https://www.realfakebirds.app',
    linkText: 'play the game',
    gitHub: 'https://github.com/rrcampbell-exe/real-fake-birds'
  },
  {
    logo: './builds/yarnsworths-tales-app-logo.png',
    alt: 'The cartoon cat Yarnsworth grins mischieviously',
    title: 'Yarnsworth\'s Tales',
    teaser: 'Trillions of tales await.',
    description: 'This storyline generator for fantasy tabletop roleplaying games inspires players at every stage of gameplay.',
    technologies: 'HTML, JavaScript, CSS, Bootstrap',
    link: 'https://rrcampbell-exe.github.io/yarnsworths-tales/',
    linkText: 'discover your next d&d adventure',
    gitHub: 'https://github.com/rrcampbell-exe/yarnsworths-tales'
  },
  {
    logo: './builds/ryan-r-campbell-app-logo.png',
    alt: 'A black-and-white image of Ryan R. Campbell gazing to the right',
    title: 'The Ryan R. Campbell Website',
    teaser: 'Books and builds and blogs, oh my!',
    description: 'The blog, portfolio, and personal site for author and technologist Ryan R. Campbell.',
    technologies: 'React, Sass, Vite, react-router, visx',
    link: 'https://ryanrcampbell.com/',
    linkText: 'return to the homepage',
    gitHub: 'https://github.com/rrcampbell-exe/ryan-r-campbell'
  },
]

export default builds
