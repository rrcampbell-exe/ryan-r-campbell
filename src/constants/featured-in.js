const featuredIn = [
  { 
    name: 'Isthmus',
    alt: "The all-capital-letters logo for Isthmus, a publication from Madison, Wisconsin",
    link: 'https://isthmus.com/arts/books/rr-campbell-imminent-dawn/',
    src: './featured-in/isthmus-logo.png',
    className: 'dark',
  },
  {
    name: 'Career Clarity',
    alt: "The logo for the Career Clarity podcast, including the words lucrative, soulful, and joyful",
    link: 'https://getcareerclarity.com/episode95/',
    src: './featured-in/Career-Clarity-Logo.png',
    className: 'dark',
  },
  {
    name: 'The Lifelong Learner',
    alt: "The logo for The Lifelong Learner, news from UW-Madison's Continuing Studies department",
    link: 'https://continuingstudies.wisc.edu/blog/isthmus-profiles-sci-fi-author-and-writers-institute-presenter-r-r-campbell/',
    src: './featured-in/Lifelong-Learner-Logo-UW.png',
    className: '',
  },
  {
    name: 'The Writescast Network',
    alt: "The logo for the Writescast Network, featuring a pencil, a microphone, and books",
    link: 'https://ryanrcampbell.com/podcasts/',
    src: './featured-in/The-Writescast-Network-Logo-No-Icons.png',
    className: 'light',
  },
  {
    name: 'The WROTE Podcast',
    alt: "The logo for the WROTE podcast, which features each letter of the word in large text",
    link: 'http://www.wrotepodcast.com/tag/r-r-campbell/',
    src: './featured-in/Wrote-Podcast-Logo.png',
    className: 'minify',
  },
  {
    name: 'Columbus Journal',
    alt: "The logo for the Columbus Journal, which is based out of Columbus, Wisconsin",
    link: 'https://wiscnews.com/news/community/columbusjournal/news/local/columbus-books-and-beer-event-hosts-sci-fi-author-r-r-campbell/article_17aa664e-3791-57d1-b708-d69688534659.html',
    src: './featured-in/Columbus-Journal-Logo.png',
    className: '',
  },
  {
    name: 'Appleton Monthly',
    alt: "The logo for Appleton Monthly, a print magazine from Wisconsin's Fox Valley",
    link: 'https://ryanrcampbell.com/2019/08/06/appleton-monthly-features-imminent-dawn/',
    src: './featured-in/Appleton-Monthly-Logo.png',
    className: 'dark',
  },
  {
    name: 'Neurotic Nourishment',
    alt: "The logo for the Neurotic Nourishment podcast, which shows host Lindsay Weisner laughing",
    link: 'https://podcasters.spotify.com/pod/show/neuroticnourishment/episodes/Season-Three--Episode-1-Suicide--Silence-and-What-Comes-Next-ep0oqa',
    src: './featured-in/Neurotic-Nourishment-Logo.png',
    className: 'minify',
  }
]

export default featuredIn
