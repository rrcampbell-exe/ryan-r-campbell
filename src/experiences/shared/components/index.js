import BlogPost from './Blog/BlogPost'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Contact from './Contact/Contact'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'

export { BlogPost, BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner }
