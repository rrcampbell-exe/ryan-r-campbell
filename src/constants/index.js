import awards from './awards'
import books from './book-data'
import builds from './builds-data'
import featuredIn from './featured-in'
import popText from './pop-text'
import reviews from './reviews'
import shortBio from './short-bio'
import boasts from './boasts.jsx'

export { awards, books, builds, featuredIn, popText, reviews, shortBio, boasts }
