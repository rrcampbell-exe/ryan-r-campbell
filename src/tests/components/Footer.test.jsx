import { describe, expect, test } from 'vitest'
import { Footer } from '../../experiences/shared/components'
import { render } from '@testing-library/react'
import { builds } from '../../constants'

describe('Footer', () => {
  test('default snapshot', () => {
    const { asFragment } = render(<Footer />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('should contain the site logo', () => {
    const { getByAltText } = render(<Footer />)
    const siteLogo = getByAltText('ryan r. campbell logo')
    expect(siteLogo).toBeInTheDocument()
  })
  test('should contain the current year', () => {
    const { getByText } = render(<Footer />)
    const currentYear = getByText(`Â© 2012 - ${new Date().getFullYear()} Ryan R. Campbell`)
    expect(currentYear).toBeInTheDocument()
  })
})
