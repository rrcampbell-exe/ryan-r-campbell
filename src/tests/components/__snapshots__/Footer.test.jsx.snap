// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > default snapshot 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="site-logo-container"
    >
      <img
        alt="ryan r. campbell logo"
        src="/favicon-2024-ryan-r-campbell.png"
      />
    </div>
    <span>
      Â© 2012 - 2024 Ryan R. Campbell
    </span>
  </footer>
</DocumentFragment>
`;
